.github/workflows/ci.yml

```diff
@@ -1,20 +1,52 @@
 name: CI Workflow
 on:
-  # Run this workflow every time a new commit pushed to upstream/fork repository.
-  # Run workflow on fork repository will help contributors find and resolve issues before sending a PR.
   push:
+    branches:
+      - master
+      - release-*
+  workflow_dispatch: {}
   pull_request:
+    branches:
+      - master
+      - release-*
 defaults:
   run:
     working-directory: go/src/sigs.k8s.io/work-api
 
+env:
+  # Common versions
+  GO_VERSION: '1.16.x'
```

Comment:
#### Comment > General
Why not 1.17, or 1.18?

Reply:
@ryanzhang-oss , 
Current go version is 1.16 and trying 1.17 makes the verify-all.sh fail.
Let's handle Go version upgrade in a separate PR.